pkg/util/util.go

```diff
@@ -0,0 +1,29 @@
+package util
+
+import (
+	"os"
+	"path/filepath"
+)
+
+// ToFileName converts a name to be alphanumeric plus underscores.
+func ToFileName(name string) string {
+	for i := 0; i < len(name); i++ {
+		if (name[i] < 'a' || name[i] > 'z') && (name[i] < 'A' || name[i] > 'Z') && (name[i] < '0' || name[i] > '9') {
+			name = name[:i] + "_" + name[i+1:]
+		}
+	}
+	return name
+}
+
+func EnsureUniqueFilename(base, name string) string {
+	for {
+		if _, err := os.Stat(filepath.Join(base, name)); err == nil {
+			name += "_"
+		} else if !os.IsNotExist(err) {
+			return ""
+		} else {
+			break
+		}
+	}
+	return name
```

Comment:
Can we do the traditional counter method? I don't know if a human would be looking at these names, but it would be difficult to see how many underscores appear in a name.
```suggestion
	var counter int
	uniqueName := name
	for {
		if _, err := os.Stat(filepath.Join(base, uniqueName)); err == nil {
			counter++
			uniqueName = strings.Sprintf("%s_%d", name, counter)
		} else if !os.IsNotExist(err) {
			return ""
		} else {
			break
		}
	}
	return uniqueName
```

Reply:
Yeah I guess a counter would be better, lol. Name collisions should be very rare, and the filenames are irrelevant to the user, but for the sake of poking around in the dataset directory, this is definitely better